libevmasm/AssemblyItem.cpp

```diff
@@ -61,20 +61,21 @@ AssemblyItem AssemblyItem::toSubAssemblyTag(size_t _subId) const
 	return r;
 }
 
-std::pair<size_t, size_t> AssemblyItem::splitForeignPushTag() const
+std::pair<SubAssemblyID, size_t> AssemblyItem::splitForeignPushTag() const
 {
 	solAssert(m_type == PushTag || m_type == Tag || m_type == RelativeJump || m_type == ConditionalRelativeJump);
```

Comment:
Based on my investigation, I think we could limit it to just `PushTag`:
```suggestion
	solAssert(m_type == PushTag);
```

And similarly in `setPushTagSubIdAndTag()` and `toSubAssemblyTag()`.

Though maybe it's better done in a separate PR. There could be other places like this and it would be good to also have assembly import validations against it.

Reply:
Makes sense, thanks for investigating this!